src/pages/Session.tsx

```diff
@@ -94,3 +110,37 @@ export default function SessionCreator() {
     </>
   );
 }
+export const getServerSideProps = (async ({ query: { type, sessionId } }) => {
+  if (type === "create") {
+    return {
+      props: {
+        FormData: {
+          student: {},
+          test: {},
+          timeline: {},
+          session: {},
+        },
+        FormType: type,
+      },
+    };
+  } else if (type === "edit" && sessionId) {
+    const FormData = await getASession(Number(sessionId));
```

Comment:
why are you doing getASession when that session data is already there with you ? why the additional instance.get() request

Reply:
I want to get the data for the session with the particular id

Reply:
but its already there. why get again?